File: eel-core/src/main/kotlin/io/eels/component/avro/avroReaderFns.kt
package io.eels.component.avro

import org.apache.avro.file.DataFileReader
import org.apache.avro.file.SeekableFileInput
import org.apache.avro.generic.GenericDatumReader
import org.apache.avro.generic.GenericRecord
import java.nio.file.Path

fun createAvroReader(path: Path): DataFileReader<GenericRecord> {
  val datumReader = GenericDatumReader<GenericRecord>()
  return DataFileReader<GenericRecord>(SeekableFileInput(path.toFile()), datumReader)
}

File: eel-components/src/main/scala/io/eels/component/avro/AvroReaderFns.scala
package io.eels.component.avro

import org.apache.avro.file.DataFileReader
import org.apache.avro.generic.{GenericDatumReader, GenericRecord}
import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.{AvroFSInput, FileSystem, Path}

object AvroReaderFns {

  def createAvroReader(path: Path)
                      (implicit conf: Configuration, fs: FileSystem): DataFileReader[GenericRecord] = {
    val datumReader = new GenericDatumReader[GenericRecord]()
    new DataFileReader[GenericRecord](new AvroFSInput(fs.open(path), fs.getFileStatus(path).getLen), datumReader)
  }
}